samples/client.c

```diff
@@ -41,9 +42,27 @@
 #include <assert.h>
 //#include <sys/uio.h>
 
-#include "../lib/muser.h"
-#include "../lib/muser_priv.h"
 #include "../lib/common.h"
+#include "../lib/muser.h"
+#include "../lib/tran_sock.h"
+
+static char *irq_to_str[] = {
+    [LM_DEV_INTX_IRQ_IDX] = "INTx",
+    [LM_DEV_MSI_IRQ_IDX] = "MSI",
+    [LM_DEV_MSIX_IRQ_IDX] = "MSI-X",
+    [LM_DEV_ERR_IRQ_INDEX] = "ERR",
+    [LM_DEV_REQ_IRQ_INDEX] = "REQ"
+};
+
+void
+lm_log(UNUSED lm_ctx_t *lm_ctx, UNUSED lm_log_lvl_t lvl, const char *fmt, ...)
```

Comment:
Unused function?

Reply:
It's called from tran_sock.c code, so we need it here.

Reply:
There is also another definition of lm_log() in lib/muser_ctx.c
As tran_sock.c is part of library, it should be using the functions defined in library.
sample/* are test examples, we should not be defining functions there and using it in library.

Reply:
tran_sock.c is used by both samples/client.c - which does not and should not link against libmuser - and libmuser.

In the first case, tran_sock.c's lm_log links with the one in samples/client.c
In the second case, tran_sock.c's lm_log links with the one in lib/muser_ctx.c

Make sense?

Reply:
Ok. Thanks got it.